File: examples/language-features/CSharp/dotnet/5.0/10.0/library/CSharp6/NameOfOperator/NameOfOperator.cs
using System;

namespace Net5_CSharp10.CSharp6.NameOfOperator
{
    public class Account
    {
        private readonly string _owner;

        public string Owner
        {
            get { return _owner; }
        }

        // nameof produces the name as a compile-time constant, so renaming the
        // parameter updates this string too — a literal "owner" would not.
        public Account(string owner)
        {
            if (owner == null)
            {
                throw new ArgumentNullException(nameof(owner));
            }

            _owner = owner;
        }

        // Only the final identifier is produced, not the qualified path.
        public static string MemberName()
        {
            return nameof(Account.Owner);
        }

        public static string TypeName()
        {
            return nameof(Account);
        }

        public static string LocalName()
        {
            int total = 0;
            return nameof(total) + total;
        }

        // A namespace works too, and yields only its last segment.
        public static string NamespaceName()
        {
            return nameof(System.Collections);
        }
    }
}

File: examples/language-features/VB.NET/dotnet/Net10/src/Vb14/NameOfOperator/NameOfOperator.vb
Option Strict On

Imports System

Namespace Vb14.NameOfOperator
    Public Class Account
        Private ReadOnly _owner As String

        Public ReadOnly Property Owner As String
            Get
                Return _owner
            End Get
        End Property

        ' NameOf produces the name as a compile-time constant, so renaming the
        ' parameter updates this string too. VB spells it NameOf; C# spells it
        ' nameof — the same operator, arriving in both languages together.
        Public Sub New(owner As String)
            If owner Is Nothing Then
                Throw New ArgumentNullException(NameOf(owner))
            End If

            _owner = owner
        End Sub

        ' Only the final identifier is produced, not the qualified path.
        Public Shared Function MemberName() As String
            Return NameOf(Account.Owner)
        End Function

        Public Shared Function TypeName() As String
            Return NameOf(Account)
        End Function

        Public Shared Function LocalName() As String
            Dim total As Integer = 0
            Return NameOf(total) & total.ToString()
        End Function
    End Class
End Namespace
